File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/CallbackResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import com.pengrad.telegrambot.model.CallbackQuery;
import org.springframework.stereotype.Component;

import java.lang.reflect.Parameter;

@Component
public class CallbackResolver implements ParameterResolver<CallbackQuery> {

    @Override
    public boolean supports(Parameter p) {
        return p.getType().equals(CallbackQuery.class);
    }

    @Override
    public CallbackQuery resolve(Parameter parameter, HandlerCtx ctx) {
        return ctx.update().callbackQuery();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/CallbackResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import com.pengrad.telegrambot.model.CallbackQuery
import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.params.CallbackParam
import java.lang.reflect.Parameter

@Component
class CallbackResolver : ParameterResolver<CallbackQuery?> {
    override fun supports(p: Parameter?): Boolean =
        p?.getType() == CallbackQuery::class.java && p.isAnnotationPresent(CallbackParam::class.java)

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): CallbackQuery? = ctx.update?.callbackQuery()

}
